### ISSUE TITLE
Cannot use string offset as an array issues

### ISSUE THREAD

**Issue by jondor** [Ariadne-CMS/ariadne#131]
PHP 7.1

```
[Thu Jun 29 12:55:34.676192 2017] [php7:notice] [pid 12117] [client 10.35.11.90:58424]
Error on line 386 in pobject::userdb.check.source in library /cms2/libs/nieuwsbrief/userdb/\n
Uncaught Error: Cannot use string offset as an array in /opt/intranet/ariadne/lib/objects/ariadne_object.php:386\n
Stack trace:\n
 #0 /opt/intranet/ariadne/lib/objects/ariadne_object.php(469): ariadne_object->saveCustomData(Object(object), '')\n
 #1 /opt/intranet/ariadne/lib/objects/ariadne_object.php(3328): ariadne_object->save('', '')\n
 #2 /opt/intranet/ariadne/lib/ar/core/pinp.php(135): ariadne_object->_save()\n
 #3 /opt/intranet/files/templates/12/93/6/_pobject.userdb.check.source.any.inc(6): ar_core_pinpSandbox->__call('_save', Array)\n
 #4 /opt/intranet/ariadne/lib/objects/ariadne_object.php(1977): {closure}(Object(pobject))\n
 #5 /opt/intranet/ariadne/lib/templates/pobject/default.phtml(48): ariadne_object->CheckConfig('userdb.check.so...', Array)\n
 #6 /opt/intranet/ariadne/lib/objects/ariadne_object.php(207): include('/opt/intranet/a...')\n
 #7 /opt/intranet/ariadne/lib/stores/mysqlstore.phtml(180): ariadne_object->call('userdb.check.so...', Array)\n
 #8 /opt/intranet/ariadne/lib/objects/ariadne_object.php(2673): mysqlstore->ca\n
, referer: http://weba1.enschede.nl/ariadne/loader.php/sites/nieuwsbrief/site/extern/?uid=JDEkMEpST3ViSjEkdjZEcjc4NEdGdTA5bXpXUlBWMWxjMQ
```
triggered by the save 

```
$sid=getvar('sid');
if (!$data->custom['none']['selection'][$sid]):
	$data->custom['none']['selection'][$sid]=$sid;
	$data->custom['none']['lastChange']=time();
	save();
endif;
```
the sg functions are on a level higher and work (the userobject is made, it's now about turning things on off)

**Comment by jondor** [comment]
This seems mostly to be due to php 7.1, I downgraded to 7.0 and it solves this specific issue (and some others)

**Comment by ylebre** [comment]
The fix is part of PR #135 - if you have any more issues concerning php7.1 please add more issues to we can tackle those as well.
